sklearn/cluster/tests/test_hierarchical.py

```diff
@@ -34,6 +34,7 @@
 from sklearn.utils.fast_dict import IntFloatDict
 from sklearn.utils.testing import assert_array_equal
 from sklearn.utils.testing import assert_warns
+from sklearn.utils.tests.test_validation import DummyMemory, WrongDummyMemory
```

Comment:
TBH, once we're using `check_memory` it's not that important to have a test in `test_hierarchical.py`. Yes, it's good to have some kind of integration testing, but I don't think we get much value out of it in this case. I'd happily see this PR without the changes to test_hierarchical and test_pipeline.